pkg/build/apis/build/validation/validation.go

```diff
@@ -607,8 +607,8 @@ func validateTrigger(trigger *buildapi.BuildTriggerPolicy, buildFrom *kapi.Objec
 
 func validateWebHook(webHook *buildapi.WebHookTrigger, fldPath *field.Path, isGeneric bool) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if len(webHook.Secret) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath.Child("secret"), ""))
+	if len(webHook.Secret) == 0 && webHook.SecretRef == nil {
```

Comment:
Are you planning to provide an `oadm migrate` command to help people transition?

Reply:
no.